docs/design/package-manager-template.md

```diff
@@ -0,0 +1,218 @@
+- Any "cache" directories where dependencies are installed locally to disk.
+
+#### File Formats and Metadata
+
+Document any specific file format requirements:
+
+- **Package file formats**: Expected formats for downloaded packages
+- **Metadata requirements**: Additional metadata files Hermeto must provide
+- **Naming conventions**: Required naming patterns for files and directories
+- **Version handling**: How different versions should be organized
+
+#### Network Requirements
+
+Describe network-related considerations:
+
+- **Registry endpoints**: URLs and APIs Hermeto needs to access
+- **Authentication**: Any authentication requirements for package registries
+- **Rate limiting**: Considerations for API rate limits
+- **Mirror support**: Support for alternative registries or mirrors
+
+### Build Environment Config
+
+Describe how the build environment should be configured to use Hermeto's pre-fetched dependencies.
+This section will form the basis of the `generate-env` and `inject-files` commands.
+
+#### Environment Variables
+
+Describe any environment variables that need to be set so that the package manager uses the
+dependencies pre-fetched by Hermeto. A table is usually sufficient:
+
+| Variable Name | Purpose | Example Value | Required |
+|---------------|---------|---------------|----------|
+| `EXAMPLE_VAR` | Points to dependency cache | `/path/to/hermeto-deps` | Yes |
+
+#### Configuration Files
+
+Describe any files that Hermeto should generate or provide to the package manager. This will form
+the basis of the `inject-files` implementation. A tree diagram can be helpful here:
+
+```
+hermeto-deps/
+├── [package-manager-name]/
+│   ├── metadata/
+│   ├── packages/
+│   └── manager-config.json
+```
+
+If needed, add sub-sections to describe specific files in detail.
+
+#### Build Process Integration (optional)
+
+If needed, describe any build process changes that are required outside of the environment
+variables and configuration file changes above.
+
+## Implementation Notes
+
+This section helps the community evaluate the maturity of the package manager. Experimental package
+managers use the `x-` prefix in their package manager name. This section should be completed prior
+to the community declaring the package manager "fully supported."
+
+### Current Limitations
+
+Document known limitations of the current implementation:
+
+- **Missing features**: Functionality not yet implemented _in Hermeto_
+- **Edge cases**: Scenarios that may not work correctly
+- **Performance considerations**: Known performance issues or bottlenecks
+- **Ecosystem considerations**: Features and discussion in the package manager ecosystem that may
+  impact Hermeto's implementation
+
+### Testing Strategy
+
+Describe how the package manager implementation is tested:
+
+- **Unit tests**: Key areas covered by unit tests
+- **Integration tests**: End-to-end testing scenarios
+- **Test data**: Sample projects and dependencies used for testing
```

Comment:
I don't think we need an explicit paragraph on this in the design template, tests are an integral part of any technical PR, so we woudln't accept just about any technical contribution without a reasonable coverage discussed as part of the review process. If it's not the case currently, any mention of tests should go to CONTRIBUTING.md instead.